pxr/usd/usdShade/testenv/testUsdShadeValidators.cpp

```diff
@@ -410,6 +412,55 @@ TestUsdShadeMaterialBindingAPIAppliedValidator()
     TF_AXIOM(errors.size() == 0);
 }
 
+void
+TestUsdShadeConnectableValidator()
+{
+    UsdValidationRegistry &registry = UsdValidationRegistry::GetInstance();
+    const UsdValidator *validator = registry.GetOrLoadValidatorByName(
+            UsdShadeValidatorNameTokens->connectableValidator);
+    TF_AXIOM(validator);
+
+    UsdStageRefPtr usdStage = UsdStage::CreateInMemory();
+
+    // Create a Material > Shader > Shader hierarchy
+    UsdShadeMaterial::Define(usdStage, SdfPath("/RootMaterial"));
+    UsdShadeShader::Define(usdStage, SdfPath("/RootMaterial/Shader"));
+    const UsdShadeShader& insideShader = UsdShadeShader::Define(usdStage, SdfPath("/RootMaterial/Shader/InsideShader"));
+
+    // Verify error that does not allow a connectable to be parented by non-container connectable
+    UsdValidationErrorVector errors = validator->Validate(insideShader.GetPrim());
```

Comment:
Might also make sense to have a Validate for "/RootMaterial/Shader" and show no errors are reported for the same.